### ISSUE TITLE
False negative for `Layout/SpaceAfterNot` cop

### ISSUE THREAD

**Issue by ymap** [rubocop/rubocop#12195]
When there's a newline between the bang operator and its operand, the cop sometimes doesn't flag the code, even though it might be more appropriate to do so.

Here's a clear example:

```ruby
!
true
```

Interestingly, when the following line is indented by two or more spaces, the cop reports the offense correctly.

```ruby
# Not reported (newline + 1 space)
!
 true
```

```ruby
# Reported (newline + 2 spaces)
!
  true
```

This cop seems to primarily focus on the position of tokens within columns and might overlook cases where tokens span multiple lines.

--------

## Expected behavior

The cop should report an offense.

## Actual behavior

No offenses are reported.

## Steps to reproduce the problem

1. Save the provided code snippet into a file named `example.rb`.
2. Execute `bundle exec rubocop --only Layout/SpaceAfterNot example.rb`.

## RuboCop version

```
$ bundle exec rubocop -V
1.56.2 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.0.2) [x86_64-linux]
  - rubocop-performance 1.19.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.23.2
```
